contracts/tenderizer/Tenderizer.sol

```diff
@@ -120,7 +125,26 @@ abstract contract Tenderizer is Initializable, ITenderizer, SelfPermit {
 
     /// @inheritdoc ITenderizer
     function rescueUnlock() external override onlyGov returns (uint256) {
-        return _unstake(address(this), node, currentPrincipal);
+        return _unstake(address(this), node, _tokensToMigrate(node));
+    }
+
+    function migrateUnlock(address _newNode) external onlyGov returns (uint256 lockID) {
+        uint256 amount = _tokensToMigrate(node);
+
+        // Check that there's no pending migration
+        uint256 pendingMigration;
+        assembly {
+            pendingMigration := sload(PENDING_MIGRATION_SLOT)
+        }
+        require(pendingMigration == 0, "PENDING_MIGRATION");
+
+        // store penging migration amount & new node
+        assembly {
+            sstore(PENDING_MIGRATION_SLOT, amount)
+            sstore(NEW_NODE_SLOT, _newNode)
+        }
+        // set new node
+        lockID = _unstake(address(this), node, amount);
```

Comment:
Given that this will work differently for Livepeer as it has a different stake migration path than unstake-withdraw-stake I'm thinking we might want to move these function bodies into internal functions in the specific implementations instead

Reply:
Hmm, since we don't really need all the functions in Livepeer, would it be better to simply declare the external functions themselves in the implementations? Rather than internal functions?